File: Api-Buen-Sabor/src/main/java/com/example/buensabor/security/dto/NuevoUsuario.java
package com.example.buensabor.security.dto;

import lombok.Data;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import java.util.HashSet;
import java.util.Set;

@Data
public class NuevoUsuario {

    @NotBlank
    private String nombre;
    @NotBlank
    private String apellido;
    private String telefono;
    @Email
    private String email;
    @NotBlank
    private String clave;
    private boolean isEnabled;
    private String tokenPassword;
    private Set<String> roles = new HashSet<>();

}

File: buen-sabor-empresa/src/app/models/nuevo-usuario.ts
export class NuevoUsuario {
    nombre: string;
    apellido: string;
    telefono: string;
    email: string;
    clave: string;
    roles: string[];

    constructor(nombre: string, apellido: string, telefono: string, email: string, clave: string, roles:string[]){
        this.nombre = nombre;
        this.apellido = apellido;
        this.telefono = telefono;
        this.email = email;
        this.clave = clave;
        this.roles = roles;
    }
}
